flixel/addons/effects/FlxWaveSprite.hx

```diff
@@ -98,26 +103,25 @@ class FlxWaveSprite extends FlxSprite
 				case TOP:
 					if (oY <= center)
 					{
-						p  = center - oY;
+						p = center - oY;
 						offset = p * calculateOffset(p);
 					}
 			}
 			
-			_flashPoint.setTo(strength + offset, oY);
-			_flashRect2.setTo(0, oY, target.frameWidth, 1);
+			if (direction == HORIZONTAL)
+			{
+				_flashPoint.setTo(strength + offset, oY);
+				_flashRect2.setTo(0, oY, target.frameWidth, 1);
+			}
+			else
+			{
+				_flashPoint.setTo(oY, strength + offset);
+				_flashRect2.setTo(oY, 0, 1, target.frameHeight);
+			}
 			pixels.copyPixels(target.framePixels, _flashRect2, _flashPoint);
 		}
-		pixels.unlock();
 		
-		if (_targetOffset == -999)
-		{
-			_targetOffset = offset;
-		}
-		else
-		{
-			if (offset == _targetOffset)
-				_time = 0;
```

Comment:
I did'nt find a use of `_targetOffset`, and whe using `FlxWaveMode.TOP` the wave wasn't updating because this `_time = 0`